src/NHibernate/Impl/MultiCriteriaImpl.cs

```diff
@@ -35,6 +35,7 @@ public partial class MultiCriteriaImpl : IMultiCriteria
 		private string cacheRegion;
 		private IResultTransformer resultTransformer;
 		private readonly IResultSetsCommand resultSetsCommand;
+		private int _timeout;
```

Comment:
For consistency with `MultiQueryImpl`, it should be `int?`.
For correctness, it should be nullable too, because `0` means "no timeout" (it is only a [recommendation](https://docs.microsoft.com/en-us/dotnet/api/system.data.common.dbcommand.commandtimeout?view=netframework-4.7.1#System_Data_Common_DbCommand_CommandTimeout) but likely most data provider will respect this), thus this added feature is going to disable the default timeout.